packages/commerce-events-collector-segments/src/getECID.ts

```diff
@@ -0,0 +1,21 @@
+// Gets the ECID from session cookie "AMCV_{AdobeOrgID}_AdobeOrg"
+
+const ECID_SESSION_REGEX = "AMCV_[A-Z0-9]*%40AdobeOrg=([^;]+)";
+
+const getECID = (): Promise<unknown> => {
+    return new Promise((resolve, reject) => {
+        let cookieForECID = document.cookie.match(ECID_SESSION_REGEX);
+        if (cookieForECID === null) {
+            setTimeout(() => {
+                cookieForECID = document.cookie.match(ECID_SESSION_REGEX);
+            }, 1500);
+        }
+        if (cookieForECID !== null) {
+            resolve(cookieForECID[1].split("|")[1]);
```

Comment:
would crap out if `cookieForECID` is undef or not a list or `cookieForECID[1]` doesn't contain `|`

Reply:
do we have all the possible expected way this value can resolve to? Since this is how it is currently showing up for me in that session cookie.

Reply:
@jonsnyder how does Alloy parse this session cookie?

Reply:
https://github.com/adobe/alloy/blob/2b2763da75d7735719950c933ab978570eda3477/src/components/Identity/createLegacyIdentity.js#L34

This is how we do it in Alloy.

Reply:
@aniham after looking at @jonsnyder comment, I opted to revert to using alloy to fetch the ECID since if the Magento instance does not have the legacy identity cookie enabled (which is the cookie that we have now) then we cannot get the ECID while alloy can, and so it is better to let alloy take care of this case.